jdaviz/app.py

```diff
@@ -1071,24 +1090,48 @@ def _get_range_subset_bounds(self, subset_state,
             return [{"name": subset_state.__class__.__name__,
                      "glue_state": subset_state.__class__.__name__,
                      "region": spec_region,
+                     "sky_region": None,  # not implemented for spectral, but include for consistancy
                      "subset_state": subset_state}]
         return spec_region
 
     def _get_multi_mask_subset_definition(self, subset_state):
+
         return [{"name": subset_state.__class__.__name__,
                  "glue_state": subset_state.__class__.__name__,
                  "region": subset_state.total_masked_first_data(),
+                 "sky_region": None,
                  "subset_state": subset_state}]
 
-    def _get_roi_subset_definition(self, subset_state):
-        # TODO: Imviz: Return sky region if link type is WCS.
+    def _get_roi_subset_definition(self, subset_state, to_sky=False):
+
+        sky_subset_region = None
+        override_wcs = None  # only for cubeviz, where gwcs is missing spatial info
+        if to_sky:
+            wcs = subset_state.xatt.parent.coords
+            if self.config == 'cubeviz':
+                parent_data = subset_state.attributes[0].parent
+                print('PARENT DATA META', parent_data.meta)
+                override_wcs = parent_data.meta["_orig_spec"].wcs
+
+        # pixel region
         roi_as_region = roi_subset_state_to_region(subset_state)
+
+        # sky region
+        roi_as_sky_region = None
+        if to_sky:
+            roi_as_sky_region = roi_subset_state_to_region(subset_state,
+                                                           to_sky=True,
+                                                           override_wcs=override_wcs)
```

Comment:
Actually when I suggested the change, I was thinking more like this. When `to_sky` is:

* `False`: Return pixel region.
* `True`: Return sky region using the native subset parent WCS (as currently implemented in `glue-astronomy`).
* WCS object: Ignore the native subset parent WCS and use the provided WCS instead (for Cubeviz workaround).


```suggestion
            roi_as_sky_region = roi_subset_state_to_region(subset_state,
                                                           to_sky=override_wcs)
```

Reply:
The way I interpreted this is - pixel region is always returned, and sky region is computed if requested and if possible. It is always possible to compute a pixel region, but because there are cases where it's not possible to compute a sky region (no wcs) and this function checks EVERY subset, i didn't want there to be a case where it failed because one of many subsets was associated with a no-wcs data.

Reply:
My thought was the same as @pllim , that the `region` key would equal the `sky_region` if available and requested and defaulting to the `pixel_region` if either or both are False. Your implementation would make more sense if users would like both regions at once, in which case we should stick to what you have here.

Reply:
it doesn't make much difference to me - should i leave as-is or change it to only having one region key that is either pixel or sky?

Reply:
I would prefer less options in the API if possible.

Reply:
either way this will be a single argument on 'get subsets' if thats what you mean by less options